src/duckling/game/drawable/auto-create-animation-dialog.component.ts

```diff
@@ -8,43 +8,113 @@ import {
-    y: number,
-    imageKey: string,
+    numFrames : number,
+    frameDimensions : Vector,
+    imageKey : string,
 }
 
 @Component({
     selector: 'dk-auto-create-animation-dialog',
+    styleUrls: ['./duckling/game/drawable/auto-create-animation-dialog.component.css'],
     template: `
-        <button
-            md-raised-button
-            type="button"
-            (click)="onAcceptClicked()">
-            Accept
-        </button>
-        <button
-            md-raised-button
-            type="button"
-            (click)="onCancelClicked()">
-            Cancel
-        </button>
+        <div class="body">
+            <dk-number-input
+                label="Number of frames"
+                [value]="numFrames"
+                (validInput)="onNumFramesInput($event)">
+            </dk-number-input>
+
+            <dk-vector-input
+                xLabel="Frame width"
+                yLabel="Frame height"
+                [value]="frameDimensions"
+                (validInput)="onDimensionInput($event)">
+            </dk-vector-input>
+
+            <dk-browse-file
+                [dialogOptions]="dialogOptions"
+                [selectedFile]="imageKey"
+                (filePicked)="onImageFilePicked($event)">
+            </dk-browse-file>
+        </div>
+
+        <div class="footer">
+            <button
+                md-button
+                type="button"
+                (click)="onAcceptClicked()">
+                Accept
+            </button>
+            <button
+                md-button
+                type="button"
+                (click)="onCancelClicked()">
+                Cancel
+            </button>
+        </div>
     `
 })
 export class AutoCreateAnimationDialogComponent {
-    constructor(private _dialogRef : MdDialogRef<AutoCreateAnimationDialogComponent>) {
+    numFrames : number = 0;
+    frameDimensions : Vector = {x: 0, y: 0};
+    imageKey : string;
+
+    constructor(private _dialogRef : MdDialogRef<AutoCreateAnimationDialogComponent>,
+                private _path : PathService,
+                private _dialog : DialogService,
+                private _assets : AssetService,
+                private _project : ProjectService) {
     }
 
     onAcceptClicked() {
         let result : AutoCreateDialogResult = {
-            x: 50,
-            y: 50,
-            imageKey: "Hello"
+            numFrames: this.numFrames,
+            frameDimensions: this.frameDimensions,
+            imageKey: this.imageKey
         };
         this._dialogRef.close(result);
     }
 
     onCancelClicked() {
         this._dialogRef.close(null);
     }
+
+    onDimensionInput(newFrameDimensions : Vector) {
+        this.frameDimensions = newFrameDimensions;
+    }
+
+    onNumFramesInput(newNumFrames : number) {
+        this.numFrames = newNumFrames;
+    }
+
+    onImageFilePicked(newImageKey : string) {
+        let homeResourceString = this._path.join(this._project.project.home, 'resources');
```

Comment:
This is a direct copy of a method in image-drawable.component. You should extract it.